### PR TITLE
Add Chord element

### PR DESCRIPTION
**Pull request by philippjfr**
As outlined in https://github.com/ioam/holoviews/issues/2137 this PR adds support for a ``Chord`` element. So far this is just a WIP, but basically this just extends the existing ``Graph`` element to lay things out in a circle and draw bezier splines between the nodes. The next step will be to allow scaling the line thickness by a value in a separate PR, which will allow scaling the chords representing the edges between nodes and then start drawing the bands around the circle.

Progress for what is supported can be tracked in: https://anaconda.org/philippjfr/chord/notebook

### CODE REVIEW

**Comment by jbednar** [comment]
Nice!  We sure need the automatic viewport padding...

**Comment by philippjfr** [comment]
Since a Chord plot is always drawn on a unit circle I've now added some default padding which can be adjusted by using the ``.redim.range`` to adjust the range of the node x and y dimensions. This means it now provides reasonable defaults, while still letting you override them (the bkcharts version hardcoded the ranges), here's the default viewport of a full ChordPlot with labels:

bokeh_plot

**Comment by philippjfr** [comment]
This PR is now feature complete, and behaves how I think it should. The upcoming 0.12.11 bokeh release will fix a bug updating the colormapping of the edges, which shouldn't be an issue as that will be released long before our next release. It's now just missing reference doc entries and plenty of unit tests.

**Comment by philippjfr** [comment]
Ready to review and merge.

File: examples/reference/elements/bokeh/Chord.ipynb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "links = pd.DataFrame(data['links'])\n",
+    "print(links.head(3))\n",
```
Comment (jlstevens) [inline examples/reference/elements/bokeh/Chord.ipynb]: I would prefer to see this cell split:

```python
links = pd.DataFrame(data['links'])
links.head(3)
```

Then:

```python
hv.Chord(links)
```

I also notice the kernel metadata is still present in the notebooks. Could you please clear that out?

Comment (philippjfr) [reply examples/reference/elements/bokeh/Chord.ipynb]: Sure.

File: holoviews/element/util.py [hunk 2]
```diff
@@ -235,6 +235,23 @@ def circular_layout(nodes):
     return (x, y, nodes)
 
 
+def quadratic_bezier(start, end, c0=(0, 0), c1=(0, 0), steps=50):
```
Comment (jlstevens) [inline holoviews/element/util.py]: Ideally this should be handled by the plotting extension. The main advantage I see of us doing this is that it helps consistency between matplotlib/bokeh. That said, plotting libraries might have more performant/robust code (though this is using numpy so it shouldn't be too slow).

File: tests/plotting/matplotlib/testplot.py [hunk 3]
```diff
@@ -9,6 +9,8 @@
 from holoviews.plotting import comms
 
 try:
+    from matplotlib import pyplot
+    pyplot.switch_backend('agg')
```
[the remark this answers is not in the document]
Comment (philippjfr) [reply tests/plotting/matplotlib/testplot.py]: Just part of shuffling tests around.

File: holoviews/element/util.py (hunk 2 above)
Comment (jlstevens) [reply holoviews/element/util.py]: That would be nice. I might actually prefer holoviews handling this if the available bezier definitions in matplotlib/bokeh don't match up.

Comment (philippjfr) [reply holoviews/element/util.py]: Afaik they do match.

**Comment by philippjfr** [comment]
Made the requested change to the notebooks and resolved the merge conflicts. Ready to merge when tests pass.

**Comment by jlstevens** [comment]
Looks good. Merging.

**Pull request merged**